### ISSUE TITLE
Destruction of record+translations fails, when model IDs are of the type "varchar(36)" (a UUID string)

### ISSUE THREAD

**Issue by john-999** [shioyama/mobility#611]
When the ID of a database record is of the type "varchar(36)" (a UUID string, with length: 36 characters), the destruction of the record with its linked translations fails (due to a borked SQL query generated by the gem).

## Context

- Using Rails, with table backend
- `gem mobility (1.2.9)`
- `gem rails (7.0.4)`
- Table names: `Issues`, `Issue_translations`
- All models have an ID of the type `varchar(36)`, meaning a (UUID) string with a length of 36 characters.
- Example model code in `my_app/app/models/issue.rb`:

```
class Issue < ApplicationRecord

  has_many :translations

  extend Mobility
  translates :name

  before_validation ObjectIdAssignment   # <--- This line generates/assigns the ID (as a UUID string)
 
end
```
- Model ID generation code in `my_app/app/models/object_id_assignment.rb`:
```
class ObjectIdAssignment

  # Before validation:
  #  Generate a (in the DB not-yet-existing) UUID ID for the object & assign it.
  #
  def self.before_validation(object)

    while object.id.nil? do
      i = SecureRandom.uuid.downcase

      if !object.class.exists?(id: i)
        object.id = i
      end
    end

  end

end
```

## Expected Behavior

In the Rails console, the following should destroy a record from the table `Issues` and all its linked records from the table `Issue_translations`:
```
> i = Issue.first
> i.destroy
```

## Actual Behavior

In the Rails console, the SQL statement for `Issue::Translation Destroy` is borked:
```
irb(main):001:0> i = Issue.first
  Issue Load (0.4ms)  SELECT `issues`.* FROM `issues` ORDER BY `issues`.`created_at` ASC, `issues`.`id` ASC LIMIT 1
=>                                               
#<Issue:0x000056225e44a170                       
...                                              
irb(main):002:0> i.destroy
  TRANSACTION (0.3ms)  BEGIN
  Issue::Translation Load (0.4ms)  SELECT `issue_translations`.* FROM `issue_translations` WHERE `issue_translations`.`issue_id` = '46e5afce-1a45-4313-ac39-55d236d78876'
  Issue::Translation Destroy (0.6ms)  DELETE FROM `issue_translations` WHERE `issue_translations`.`` IS NULL
  TRANSACTION (0.1ms)  ROLLBACK                  
/home/user/.rbenv/versions/3.0.1/lib/ruby/gems/3.0.0/gems/mysql2-0.5.4/lib/mysql2/client.rb:148:in `_query': Mysql2::Error: Unknown column 'issue_translations.' in 'where clause' (ActiveRecord::StatementInvalid)                                          
/home/user/.rbenv/versions/3.0.1/lib/ruby/gems/3.0.0/gems/mysql2-0.5.4/lib/mysql2/client.rb:148:in `_query': Unknown column 'issue_translations.' in 'where clause' (Mysql2::Error)
```

**Comment by shioyama** [comment]
Thanks for posting this. Could you write a failing test for it? You should be able to add a column in `spec/active_record/schema.rb` and test mysql specifically by passing a `db` option to `context`, like here:

https://github.com/shioyama/mobility/blob/bc5d7a7a71136cfb783ebd1fdb1497495f347aa2/spec/generators/rails/mobility/translations_generator_spec.rb#L91-L93

**Comment by john-999** [comment]
As much as I'd like to be able to do this: I'm just a hobbyist using Rails in a very basic way (I'm not a web developer or programmer). And I have never written a Test and know how to use Github only for reporting bugs. So I'm sure any input from my side would not be helpful, sorry...

**Comment by shioyama** [comment]
In that case I need to close this. If anyone else encounters this issue, please feel free to follow up and I can re-open it.

**Comment by shioyama** [comment]
Hmm, I really don't have the knowledge to help with this. If someone can create a script/test that reproduces the issue that would help.

**Comment by KonstantinReido** [comment]
@cercxtrova Do I understand correctly that when creating the translation tables, you set UUID as the type for `translatable_id`? It looks like you’re trying to add a translation for a model whose ID is **integer**.

I’ve encountered a similar issue before. In my case, I tried changing `translatable_id` to string to allow storing translations for models with both integer and UUID IDs. However, this approach leads to a similar error.

If you’re interested in the solution I ended up using, let me know, and I’ll describe it.
